### ISSUE TITLE
[BUG] Cannot update packages

### ISSUE THREAD

**Issue by emptyVoid** [msys2/MSYS2-packages#2021]
## Describe the bug

MSYS2 base package no longer updates:
```
MSYS ~
$ pacman -Syu --noconfirm
:: Synchronizing package databases...
 mingw32                  467.1 KiB   439K/s 00:01 [#####################] 100%
 mingw32.sig              119.0   B  0.00B/s 00:00 [#####################] 100%
 mingw64                  469.7 KiB  2.40M/s 00:00 [#####################] 100%
 mingw64.sig              119.0   B  0.00B/s 00:00 [#####################] 100%
 msys                     187.6 KiB  11.5M/s 00:00 [#####################] 100%
 msys.sig                 119.0   B  0.00B/s 00:00 [#####################] 100%
:: Starting core system upgrade...
warning: terminate other MSYS2 programs before proceeding
resolving dependencies...
looking for conflicting packages...
error: failed to prepare transaction (could not satisfy dependencies)
:: installing filesystem (2020.02-3) breaks dependency 'msys2-base' required by dash
```

## Steps to Reproduce the Problem

1. Run `msys2.exe`
2. Execute `pacman -Syu --noconfirm`
3. Catch the error

Expected behavior: the core system upgrade should complete without errors.

## Additional Context: Operating System, Screenshots

- OS: Windows 10 Pro version 2004 64-bit
- MSYS2: msys2-base-x86_64-20190524.tar.xz and msys2-base-x86_64-20200602.tar.xz

**Comment by Biswa96** [comment]
Warning: Try this if you don't care about your MSYS2 installation to be nuked :)

1. Download the .tar.xz file from here https://packages.msys2.org/package/filesystem.
2. Assume that file is downloaded in `C:\MyFiles` folder. Run this command `pacman -Udd /c/MyFiles/filesystem-2020.02-3-x86_64.pkg.tar.xz`.
3. Kill every msys2 shell/program.
4. Run `pacman -Syu` again.

**Comment by lissyx** [comment]
> Warning: Try this if you don't care about your MSYS2 installation to be nuked :)
> 
>     1. Download the .tar.xz file from here https://packages.msys2.org/package/filesystem.
> 
>     2. Assume that file is downloaded in `C:\MyFiles` folder. Run this command `pacman -Udd /c/MyFiles/filesystem-2020.02-3-x86_64.pkg.tar.xz`.
> 
>     3. Kill every msys2 shell/program.
> 
>     4. Run `pacman -Syu` again.

Same repro here, and we start a msys2 from scratch on new VMs (so no `pacman -Udd`).

**Comment by lissyx** [comment]
> > Warning: Try this if you don't care about your MSYS2 installation to be nuked :)
> > ```
> > 1. Download the .tar.xz file from here https://packages.msys2.org/package/filesystem.
> > 
> > 2. Assume that file is downloaded in `C:\MyFiles` folder. Run this command `pacman -Udd /c/MyFiles/filesystem-2020.02-3-x86_64.pkg.tar.xz`.
> > 
> > 3. Kill every msys2 shell/program.
> > 
> > 4. Run `pacman -Syu` again.
> > ```
> 
> Same repro here, and we start a msys2 from scratch on new VMs (so no `pacman -Udd`):
> 
> ```
>     - >-
>       "C:\Program Files\7-zip\7z.exe" x -txz -so msys2-base-x86_64.tar.xz |
>       "C:\Program Files\7-zip\7z.exe" x -o%USERPROFILE% -ttar -aoa -si
>     - >-
>       .\msys64\usr\bin\bash.exe --login -cx "export THIS_BASH_PID=$$; ps -ef |
>       grep '[?]' | awk '{print $2}' | grep -v $THIS_BASH_PID | xargs -r kill;
>       exit 0"
>     - .\msys64\usr\bin\bash.exe --login -cx "pacman -Syu --noconfirm"
> ```

Correcting myself, placing the `filesystem-2020.02-3` file and running `pacman -Udd` prior seems to do the trick.

**Comment by lazka** [comment]
See #2022

**Comment by GitHubAdi** [comment]
Same problem on my side with a clean install after downloading msys2-base-x86_64-20200602.tar.xz from http://repo.msys2.org/distrib/x86_64/

Pacman -Syuu gives the same filesystem breaks dependency error.

I add previously installed another clean install in the morning, limited to just a couple of "Pacman -Syuu", without downloading any other packages; and this one could redo "Pacman -Syuu" and install the filesystem package after the repo update.

**Comment by DaveThornton** [comment]
Fixed it for me also!! Thanks

**Comment by tomyri** [comment]
`pacman -Sydd filesystem; pacman -Su` works also.

**Comment by marler8997** [comment]
workaround fix appears to be here: https://github.com/msys2/MSYS2-packages/pull/2022

**Comment by marler8997** [comment]
This broke ziglang's azure pipelines that use MSYS2.  Tried @Biswa96 suggestion of downloading the `filesystem` package archive and installing from that, but it failed because it conflicted with the pre-existing `filesystem` package.  @tomyri suggestion of force upgrading the `filesystem` pacakge with dependency checking disabled seemed to work.

see https://github.com/ziglang/zig/pull/5688

**Comment by Jimmy-Z** [comment]
I'd like to re-post here that `pacman -S dash` before `pacman -Syu` is a simpler (and safer) workaround instead of force upgrade filesystem, as https://github.com/msys2/MSYS2-packages/pull/2022#issuecomment-648917345.

**Comment by RobertBColton** [comment]
@Jimmy-Z that did not seem to work for me.

**Comment by Jimmy-Z** [comment]
@RobertBColton works on my desktop and my laptop. No idea why it doesn't work for you though. And, you should probably share some details, like which exact step doesn't work, and the output could help too.

**Comment by RobertBColton** [comment]
@Jimmy-Z Azure build below, we're checking it out fresh. It may look different because I keep committing different variations of what you guys propose here, nothing's cutting it though.
https://github.com/enigma-dev/RadialGM/pull/134/files#diff-3364e2ba7fdae37fba56fa0d6d765896R63

**Comment by jbruchon** [comment]
Manually installing the filesystem tar.xz worked for me.

**Comment by RobertBColton** [comment]
Is it necessary to download that tar for the fixes @Jimmy-Z and stuff mentioned?

**Comment by emptyVoid** [comment]
@RobertBColton, you could possibly use the same workaround implemented in `vcpkg` -- remove it before making the core system upgrade (if you don't need `dash` of course), e.g.:
```
%CD:~0,2%\msys64\usr\bin\pacman --noconfirm -Rc dash
%CD:~0,2%\msys64\usr\bin\pacman --noconfirm -Syu
...
```

**Comment by RobertBColton** [comment]
Hey, that did do the trick! Thanks!

**Comment by Jimmy-Z** [comment]
@RobertBColton it could be that your system is too far behind that dash needs a newer filesystem, that again brought up the problem, you should be able to install dash afterwards I assume.

**Comment by DDoSolitary** [comment]
I followed @Jimmy-Z's suggestion and the upgrade completed successfully. However, after that I can no longer run any msys executables (including bash, pacman, etc), getting the following error:

```
      0 [main] pacman (16444) shared_info::initialize: size of shared memory region changed from 51128 to 40888
```

**Comment by trini** [comment]
It looks like the underlying problem has been resolved and you can go back to installing+updating as before.  At least, my Azure CI jobs are once again completing and I re-ran the failed ones and they work now.

**Comment by emptyVoid** [comment]
@DDoSolitary, have you tried killing `gpg-agent` and `dirmngr` with Task Manager, or restarting your PC?
This kind of messages usually appear when some of the `MSYS2`-spawned processes keeps running preventing the core system upgrade from completing.

**Comment by DDoSolitary** [comment]
@emptyVoid Yeah I think that is the problem. I got it working by restarting the computer.

**Comment by goyalyashpal** [comment]
is this fixed, can this be closed?

**Comment by Jimmy-Z** [comment]
I believe this was fixed by pr #2022
